src/Events/Admin/Notice/Rest_Api.php

```diff
@@ -96,23 +99,16 @@ public function is_rest_api_blocked( bool $force = false ): bool {
 			return ! empty( $this->blocked_endpoint );
 		}
 
-		$v1_api    = tribe( 'tec.rest-v1.main' );
-		$event_api = get_rest_url( null, $v1_api->get_events_route_namespace() );
-		$response  = wp_remote_get( $event_api );
-		if ( is_wp_error( $response ) || wp_remote_retrieve_response_code( $response ) !== 200 ) {
-			$this->blocked_endpoint = $event_api;
-			tec_timed_option()->set( $cache_key, $event_api, $cache_timeout );
-
-			return true;
-		}
-
-		$views_api = get_rest_url( null, V2::ROOT_NAMESPACE );
-		$response  = wp_remote_get( $views_api );
-		if ( is_wp_error( $response ) || wp_remote_retrieve_response_code( $response ) !== 200 ) {
-			$this->blocked_endpoint = $views_api;
-			tec_timed_option()->set( $cache_key, $views_api, $cache_timeout );
+		// Check multiple endpoints to determine if the REST API is blocked.
+		$endpoints = $this->get_routes_to_check();
+		foreach ( $endpoints as $endpoint ) {
+			$response = wp_safe_remote_get( $endpoint );
```

Comment:
Consider the timeout as well please

Reply:
@dpanta94 Could you elaborate?

Reply:
Yes! I was talking about the `timeout` offset being passed into `wp_sage_remote_get` through the `$endpoint` variable. By default, if not specified otherwise its `5`. I think we could go to `3`. Not too important, i just wanted to make sure we are considering it.